## Heroes Of Pymoli - Final Report 

In [1]:
# Dependencies and Setup
import pandas as pd

# File to Load (Remember to Change These)
file_to_load = "Resources/purchase_data.csv"

# Read Purchasing File and store into Pandas data frame
purchase_data = pd.read_csv(file_to_load)
purchase_data.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

## Player Count

In [2]:
# Find total number of players
total_players = len(purchase_data['SN'].unique())

# Display a dataframe
total_players_df = pd.DataFrame({'Total Players':[total_players]})
total_players_df

Total Players
0            576

## Purchasing Analysis (Total)

In [3]:
# Find columns Item ID, Purchase ID, Price
unique_items = purchase_data.loc[:, ["Item ID","Purchase ID","Price"]]

# Obtain number of unique items
items = len(unique_items['Purchase ID'].unique())

# Calculate average price
avg_price = unique_items['Price'].mean()

#Find total number of purchase
number_of_purchase = purchase_data['Purchase ID'].count()

# Find total revenue
total_revenue = purchase_data['Price'].sum()

# Create a dataframe
summary_df = pd.DataFrame ({'Number of Unique Items':[items],
                            'Average Price':[avg_price],
                            'Number of Purchases':[number_of_purchase],
                            'Total Revenue':[total_revenue]})

# Formatting Average Price, Total Revenue
summary_df["Average Price"] = summary_df["Average Price"].map("${:.2f}".format)
summary_df["Total Revenue"] = summary_df["Total Revenue"].map("${:,.2f}".format)

summary_df

Number of Unique Items Average Price  Number of Purchases Total Revenue
0                     780         $3.05                  780     $2,379.77

## Gender Demographics

In [4]:
#Group data by Gender and filter duplicates
grouped_gender = purchase_data.groupby(['Gender'])
unique_gender = grouped_gender.nunique()

#Total Gender
total_gender = unique_gender['SN'].sum()

#Percentage and Count of Players
count = unique_gender["SN"].unique()
percentage = unique_gender["SN"] * 100/ total_gender

# Create a dataframe
gender_df = pd.DataFrame({'Total Count':count,
                           'Percentage of Players':percentage})

# Formmatting
gender_df["Percentage of Players"] = gender_df["Percentage of Players"].map("{:.2f}%".format)

# Sorting the data by Total Count
gender_df = gender_df.sort_values('Total Count', ascending=False)

gender_df

Total Count Percentage of Players
Gender                                                  
Male                           484                84.03%
Female                          81                14.06%
Other / Non-Disclosed           11                 1.91%

In [8]:
# Export DF to excel
gender_df.to_excel(r'Final_output\gender_demographics.xlsx', index = True)


## Purchasing Analysis (Gender)

In [5]:
# Find number of purchase by gender
purchase_count = purchase_data['Gender'].value_counts()

# Find average purchase per gender
avg_purchase_price = grouped_gender['Price'].mean()

# Find total purchase per gender
total_purchase_value = grouped_gender['Price'].sum()

# Find average total purchase per person 
purchase_per_person = total_purchase_value / count

# Create a dataframe
purchasing_df =  pd.DataFrame({"Purchase Count":purchase_count,
                               "Average Purchase Price":avg_purchase_price,
                               "Total Purchase Value":total_purchase_value,
                               "Avg Total Purchase per Person":purchase_per_person})

# Formatting columns
purchasing_df["Average Purchase Price"] = purchasing_df["Average Purchase Price"].map("${:.2f}".format)
purchasing_df["Total Purchase Value"] = purchasing_df["Total Purchase Value"].map("${:,.2f}".format)
purchasing_df["Avg Total Purchase per Person"] = purchasing_df["Avg Total Purchase per Person"].map("${:.2f}".format)

purchasing_df



Purchase Count Average Purchase Price  \
Female                            113                  $3.20   
Male                              652                  $3.02   
Other / Non-Disclosed              15                  $3.35   

                      Total Purchase Value Avg Total Purchase per Person  
Female                             $361.94                         $4.47  
Male                             $1,967.64                         $4.07  
Other / Non-Disclosed               $50.19                         $4.56

In [9]:
# Export DF to excel
purchasing_df.to_excel(r'Final_output\purchasing.xlsx', index = True)

## Age Demographics

In [11]:
# Set bins and bin labels
age_bins = [0, 9, 14, 19, 24, 29, 34, 39, 110]
group_names = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "35-39", "40+"]

# Add "age ranges" into bins
purchase_data["Age Ranges"] = pd.cut(purchase_data["Age"], age_bins, labels=group_names)

# Group by 'Age Ranges'
groupedby_ages = purchase_data.groupby('Age Ranges')

# Find number of unique players by age
age_counts = groupedby_ages['SN'].nunique()  

# Find total of unique players
total_unique_players = age_counts.sum()

# Find percentage of unique players
players_percent = age_counts *100 / total_unique_players

# Create a dataframe
age_demographics_df = pd.DataFrame ({"Total Count":age_counts,
                                     'Percentage of Players':players_percent})

# Formatting column Percentange of Players
age_demographics_df["Percentage of Players"] = age_demographics_df["Percentage of Players"].map("{:.2f}%".format)

age_demographics_df



Total Count Percentage of Players
Age Ranges                                   
<10                  17                 2.95%
10-14                22                 3.82%
15-19               107                18.58%
20-24               258                44.79%
25-29                77                13.37%
30-34                52                 9.03%
35-39                31                 5.38%
40+                  12                 2.08%

In [12]:
# Export DF to excel
age_demographics_df.to_excel(r'Final_output\age_demographics.xlsx', index = True)

## Purchasing Analysis (Age)

In [14]:
# Using previous group by "Age Range" and "age_counts", make calculations
purchase_count = groupedby_ages['Item ID'].count() 
avg_puchase_price = groupedby_ages['Price'].mean() 
total_purchase_value = groupedby_ages['Price'].sum() 
avg_unique_person = total_purchase_value / age_counts 

# Create a dataframe
age_purchasing_df =  pd.DataFrame({ "Purchase Count":purchase_count,
                                    "Average Purchase Price":avg_puchase_price,   
                                    "Total Purchase Value":total_purchase_value,
                                    "Avg Total Purchase per Person":avg_unique_person})

# Formatting
age_purchasing_df["Average Purchase Price"] = age_purchasing_df["Average Purchase Price"].map("${:.2f}".format)
age_purchasing_df["Total Purchase Value"] = age_purchasing_df["Total Purchase Value"].map("${:,.2f}".format)
age_purchasing_df["Avg Total Purchase per Person"] = age_purchasing_df["Avg Total Purchase per Person"].map("${:.2f}".format)

age_purchasing_df

Purchase Count Average Purchase Price Total Purchase Value  \
Age Ranges                                                               
<10                     23                  $3.35               $77.13   
10-14                   28                  $2.96               $82.78   
15-19                  136                  $3.04              $412.89   
20-24                  365                  $3.05            $1,114.06   
25-29                  101                  $2.90              $293.00   
30-34                   73                  $2.93              $214.00   
35-39                   41                  $3.60              $147.67   
40+                     13                  $2.94               $38.24   

           Avg Total Purchase per Person  
Age Ranges                                
<10                                $4.54  
10-14                              $3.76  
15-19                              $3.86  
20-24                              $4.32  
25-29                              $3.81  
30-34                              $4.12  
35-39                              $4.76  
40+                                $3.19

In [15]:
# Export DF to excel
age_purchasing_df.to_excel(r'Final_output\age_purchasing.xlsx', index = True)

## Top Spenders

In [16]:
# Group by "SN"
groupedby_sn = purchase_data.groupby("SN")

# Find number of purchase per SN
purchase_count_sn = groupedby_sn['Price'].count()

#Find average purchase per SN
avg_price = groupedby_sn['Price'].mean()

# Find total purchase per SN
total_price = groupedby_sn['Price'].sum()

# Create dataframe
top_spender_df = pd.DataFrame({ 'Purchase Count': purchase_count_sn,
                              'Average Purchase Price': avg_price,
                              'Total Purchase Value': total_price})

# Sorting by Purchase Value in descending order
top_spender_final = top_spender_df.sort_values('Total Purchase Value', ascending=False)

# Formatting
top_spender_final['Average Purchase Price'] = top_spender_final['Average Purchase Price'].map("${:.2f}".format)
top_spender_final['Total Purchase Value'] = top_spender_final['Total Purchase Value'].map("${:.2f}".format)


top_spender_final.head(5)


Purchase Count Average Purchase Price Total Purchase Value
SN                                                                     
Lisosia93                 5                  $3.79               $18.96
Idastidru52               4                  $3.86               $15.45
Chamjask73                3                  $4.61               $13.83
Iral74                    4                  $3.40               $13.62
Iskadarya95               3                  $4.37               $13.10

In [17]:
# Export DF to excel
top_spender_df.to_excel(r'Final_output\top_spender.xlsx', index = True)

## Most Popular Items

In [11]:
# Group by "SN"
grouped_df = purchase_data.groupby(["Item ID", "Item Name"])

# Find number of purchase per SN
purchase_count = grouped_df['Price'].count()

#Find average purchase per SN
avg_price = grouped_df['Price'].mean()

# Find total purchase per SN
total_price = grouped_df['Price'].sum()

# Create dataframe
most_pop_item = pd.DataFrame({'Purchase Count': purchase_count,
                              'Average Purchase Price': avg_price,
                              'Total Purchase Value': total_price})

# Sorting by Purchase Value in descending order
most_pop_item_final = most_pop_item.sort_values('Purchase Count', ascending=False)

# Formatting
most_pop_item_final['Average Purchase Price'] = most_pop_item_final['Average Purchase Price'].map("${:.2f}".format)
most_pop_item_final['Total Purchase Value'] = most_pop_item_final['Total Purchase Value'].map("${:.2f}".format)

most_pop_item_final.head(5)

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                     Average Purchase Price  \
Item ID Item Name                                                             
92      Final Critic                                                  $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm                  $4.23   
145     Fiery Glass Crusader                                          $4.58   
132     Persuasion                                                    $3.22   
108     Extraction, Quickblade Of Trembling Hands                     $3.53   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
145     Fiery Glass Crusader                                       $41.22  
132     Persuasion                                                 $28.99  
108     Extraction, Quickblade Of Trembling Hands                  $31.77

## Most Profitable Items

In [13]:
# Sorting by Total Purchase Value in descending order (using the previous "most_pop_item" dataframe)
most_profitable_item = most_pop_item.sort_values('Total Purchase Value', ascending=False)

# Formatting
most_profitable_item['Average Purchase Price'] = most_profitable_item['Average Purchase Price'].map("${:.2f}".format)
most_profitable_item['Total Purchase Value'] = most_profitable_item['Total Purchase Value'].map("${:.2f}".format)

most_profitable_item.head(5)

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                     Average Purchase Price  \
Item ID Item Name                                                             
92      Final Critic                                                  $4.61   
178     Oathbreaker, Last Hope of the Breaking Storm                  $4.23   
82      Nirvana                                                       $4.90   
145     Fiery Glass Crusader                                          $4.58   
103     Singed Scalpel                                                $4.35   

                                                     Total Purchase Value  
Item ID Item Name                                                          
92      Final Critic                                               $59.99  
178     Oathbreaker, Last Hope of the Breaking Storm               $50.76  
82      Nirvana                                                    $44.10  
145     Fiery Glass Crusader                                       $41.22  
103     Singed Scalpel                                             $34.80

## Final Analysis

1. Males reprented 84% of all customers and they also made about 84% of all purchases. However, they purchased items with an average price about 9% lower than purchased by females and other/non-disclosed customers.

2. 20-24-year-old customers made the most in purchases (365 purchases), they spend 3.05 dollars on average. While 35-39-year old bought items with an average purchase price 18% higher, these group accounted for only 41 purchases, which represents  only 8% of total sales. 10 years old or younger customers also tend to purchase items with higher average purchase price - 10% higher on average and they account for only 4% of the total sales. The worst sales count come from customers at the age of 40 years old or older, they account for just 2.2% of the sales, however they spent 2.94 dollars on average - nothing bad.

3.  There is a positive correlation between the most popular and the most probitable item, which makes sense since its average price is attactive.